lib/date_time/date_time_core.c

```diff
@@ -53,39 +56,46 @@ static void date_time_core_notify_event(enum date_time_evt_type time_source)
 	}
 }
 
-static void date_time_core_schedule_update(bool check_pending)
+static void date_time_core_schedule_update(bool fast)
 {
-	/* (Re-)schedule time update work
-	 * if periodic updates are requested in the configuration.
-	 */
-	if (CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS > 0) {
-		/* Don't reschedule time update work in some cases,
-		 * e.g. if time is not obtained and the work is already pending.
-		 */
-		if (check_pending && k_work_delayable_is_pending(&date_time_update_work)) {
-			return;
-		}
+	int delay = CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS;
+
+	/* If we have already exhausted the maximum fast retries, switch to non-fast. */
+	if (fast && atomic_inc(&fast_retry_count) >= CONFIG_DATE_TIME_MAX_FAST_RETRIES) {
+		fast = false;
+	}
 
-		LOG_DBG("New periodic date time update in: %d seconds",
-			CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS);
+	/* Only (Re-)schedule time update work
+	 * if periodic updates are requested in the configuration, or this is a
+	 * fast retry.
+	 */
+	if (!fast && CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS <= 0) {
+		return;
+	}
 
-		k_work_reschedule_for_queue(
-			&date_time_work_q,
-			&date_time_update_work,
-			K_SECONDS(CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS));
+	if (fast) {
+		delay = CONFIG_DATE_TIME_FAST_RETRY_INTERVAL_SECONDS;
+	} else {
+		/* Reset the fast retry counter since this will be a non-fast retry. */
+		atomic_set(&fast_retry_count, 0);
 	}
+
+	LOG_DBG("New periodic date time update in: %d seconds%s", delay, fast?" (fast retry)":"");
```

Comment:
```suggestion
	LOG_DBG("New periodic date time update in: %d seconds%s", delay, fast ? " (fast retry)" : "");
```

not sure if this is an actual formatting issue or just personal pref.

Reply:
It's both! My personal preferences do not align with our coding guidelines on this front 😆. Good catch, thanks, will correct it